### ISSUE TITLE
False positive on await and None

### ISSUE THREAD

**Issue by lmazuel** [python/mypy#7587]
Hello,

With the following code:
```python
def foo() -> Union[None, Awaitable[None]]:
   # might return an awaitable or nothing

result = foo()
if hasattr(result, '__await__'):
    await result
```
I get a `error: Incompatible types in "await" (actual type "Optional[Awaitable[None]]", expected type "Awaitable[Any]")`

It seems to be that being that if `result` is None, this can't pass the test, mypy should know that result is not None. Or I'm missing something?

With this code the error is gone:
```python
result = foo()
if result is not None and hasattr(result, '__await__'):
    await result
```

```shell
> mypy --version                                
mypy 0.720
```

**Comment by JelleZijlstra** [comment]
This is because mypy doesn't use `hasattr()` calls to narrow types, nothing to do with `await` specifically.

**Comment by lmazuel** [comment]
So duplicate of https://github.com/python/mypy/issues/1424 for you?

**Comment by JelleZijlstra** [comment]
Yes, thanks for finding that. Going to close this issue as a duplicate.
